src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java

```diff
@@ -524,30 +525,36 @@ protected void handleResponse(Response response, Request<Message> request)
      */
     default RestAction<Message> sendFile(InputStream data, String fileName, Message message)
     {
-        Args.notNull(data, "data InputStream");
-        Args.notNull(fileName, "fileName");
+        Checks.notNull(data, "data InputStream");
+        Checks.notNull(fileName, "fileName");
 
         Route.CompiledRoute route = Route.Messages.SEND_MESSAGE.compile(getId());
-        MultipartBody body = Unirest.post(Requester.DISCORD_API_PREFIX + route.getCompiledRoute())
-                .fields(null); //We use this to change from an HttpRequest to a MultipartBody
-
-        body.field("file", data, fileName);
+        MultipartBody.Builder builder = new okhttp3.MultipartBody.Builder()
+                .setType(MultipartBody.FORM);
+        try
+        {
+            builder.addFormDataPart("file", fileName, RequestBody.create(MediaType.parse("application/octet-stream"), ByteString.read(data, Integer.MAX_VALUE)));
+        }
+        catch (IOException e)
+        {
+            return new RestAction.FailedRestAction<Message>(e);
+        }
```

Comment:
In that case this would be the only RestAction that could throw an Exception. That's why I decided to use the error consumer the user already provides instead.